libraries/discriminator/Cargo.toml

```diff
@@ -8,6 +8,7 @@ license = "Apache-2.0"
 edition = "2021"
 
 [dependencies]
+borsh = "0.10"
```

Comment:
In general, borsh adds a lot of bloat in the final binary, so we should avoid deriving it unnecessarily.

Even though features can be hard to reason about, let's featurize the borsh bits for now, similar to `type-length-value`.